### ISSUE TITLE
[improvement] Add some pagination to `Select<T>`

### ISSUE THREAD

**Issue by romancitodev** [fadeevab/cliclack#79]
It would be a good option to add some pagination in `Select` because maybe you have a list of more than 20 items and it could be chaotic to see what you are selecting.

This is a good example to take reference:
![image](https://github.com/user-attachments/assets/fb6763e4-3950-4e5c-a35c-5fc60b3690a6)

**Comment by fadeevab** [comment]
@cylewitruk This thing is still relevant! :)

**Comment by cylewitruk** [comment]
> @cylewitruk This thing is still relevant! :)

@fadeevab yeah, just been swamped as of late 😬 If you (or anyone else) have/has the bandwidth feel free to take it on in the meantime 😊

**Comment by BjornTheProgrammer** [comment]
I've decided to take a shot at it, I made pull request #83, which should fix this issue and is based upon #73.

**Comment by fadeevab** [comment]
@BjornTheProgrammer Thanks! I just noticed your message. I'll check it later.

**Comment by fadeevab** [comment]
PR #91 (based on PR #83) solves this request.
